modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java

```diff
@@ -107,20 +106,35 @@ public MuleEvent process(MuleEvent event) throws MuleException
     }
 
 
-    private void initializeWsdlRetrievalStrategy()
+    private void initializeWsdlRetrievalStrategy() throws InitialisationException
     {
         HttpRequesterConfig httpRequesterConfig = config.getConnectorConfig();
+        String url = config.getWsdlLocation();
+
+        boolean isHttpRequester = verifyIsHttpRequester(url);
+
+        if (httpRequesterConfig == null && config.isUseConnectorToRetrieveWsdl())
+        {
+            throw new InitialisationException(CoreMessages.createStaticMessage("The useConnectorToRetrieveWsdl option requires connectorConfig to work"),
```

Comment:
It would be better to print a warning message here instead of failing because the user might not want to use it everywhere (the system property affects all ws consumers). It's not necessary to test that the warning is shown.